middlewares/sponsor.middleware.js

```diff
@@ -11,17 +11,29 @@ const Constants = {
     General: require("../constants/general.constant"),
     Error: require("../constants/error.constant"),
 };
+const Sponsor = require("../models/sponsor.model");
 
 /**
  * @function parsePatch
  * @param {body: {id: ObjectId}} req 
  * @param {*} res 
  * @param {(err?) => void} next 
  * @return {void}
- * @description Delete the req.body.id that was added by the validation of route parameter.
+ * @description Delete the req.body.id that was added by the validation of route parameter, and put in sponsorDetails
```

Comment:
This no longer does the following: `Delete the req.body.id that was added by the validation of route parameter`